fl4health/clients/basic_client.py

```diff
@@ -1206,3 +1276,55 @@ def transform_gradients(self, losses: TrainingLosses) -> None:
             losses (TrainingLosses): The losses object from the train step
         """
         pass
+
+    def save_checkpoint(self) -> None:
```

Comment:
This is a bit generic and might be misleading for a user in the sense that this is actually a very specific type of checkpointing. Maybe we call this something like `save_client_state` in a future refactor?